src/libstd/sys/cloudabi/shims/fs.rs

```diff
@@ -140,6 +140,10 @@ impl Iterator for ReadDir {
     fn next(&mut self) -> Option<io::Result<DirEntry>> {
         match self.0 {}
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, Some(0))
```

Comment:
This type is uninhabited - does it really need a size_hint override?

Reply:
Yeah that makes sense to remove it. I just ported the old pr :D